ksqldb-engine/src/test/java/io/confluent/ksql/integration/WindowingSharedRuntimeIntTest.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.integration;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.common.utils.IntegrationTest;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.metastore.model.DataSource;
+import io.confluent.ksql.name.SourceName;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.services.KafkaTopicClient.TopicCleanupPolicy;
+import io.confluent.ksql.test.util.KsqlIdentifierTestUtil;
+import io.confluent.ksql.test.util.TopicTestUtil;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.OrderDataProvider;
+import io.confluent.ksql.util.QueryMetadata;
+import kafka.zookeeper.ZooKeeperClientException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.streams.kstream.SessionWindowedDeserializer;
+import org.apache.kafka.streams.kstream.TimeWindowedDeserializer;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.SessionWindow;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static io.confluent.ksql.GenericRow.genericRow;
+import static io.confluent.ksql.serde.FormatFactory.JSON;
+import static io.confluent.ksql.serde.FormatFactory.KAFKA;
+import static io.confluent.ksql.test.util.AssertEventually.assertThatEventually;
+import static io.confluent.ksql.test.util.ConsumerTestUtil.hasUniqueRecords;
+import static io.confluent.ksql.test.util.MapMatchers.mapHasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+@Category({IntegrationTest.class})
+public class WindowingSharedRuntimeIntTest {
```

Comment:
Can we just parametrize the other test vs adding an entire new test? Is it too complicated or something? Just wondering if that might be easier -- also keep in mind, adding new integration tests is really what ultimately impacts the test run time in a significant way, since it often has a nontrivial startup time to bring up the cluster/test harness/etc.

Not sure what the actual overhead for this specific test is, just wanted to point that out in case it's possible to avoid adding minutes to the time it takes to run all the tests (which is already wayyy too long lol)

Reply:
I tired. It doesn't work with the IntegrationTestHarness. When we turn it on by default we can remove the old test classes. The new test classes are quite a bit faster tho!